From 6fd08cbfa13257217032eb95bf8d8afea72e2cdb Mon Sep 17 00:00:00 2001
Message-ID: <6fd08cbfa13257217032eb95bf8d8afea72e2cdb.1758550978.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Sat, 15 Apr 2023 17:53:27 +0200
Subject: [PATCH] fix test_*font

Depending on build options and system, noto fonts may or may not be
present. Thus font tests expecting them may fail for the wrong reason.

Adjust the tests to use the fallback font instead.
---
 tests/test_general.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/test_general.py b/tests/test_general.py
index 8091023e..5d6767ce 100644
--- a/tests/test_general.py
+++ b/tests/test_general.py
@@ -183,9 +183,9 @@ def test_bug1971():
         assert doc.is_closed
 
 def test_default_font():
-    f = pymupdf.Font()
-    assert str(f) == "Font('Noto Serif Regular')"
-    assert repr(f) == "Font('Noto Serif Regular')"
+    f = pymupdf.Font(ordering=0)
+    assert str(f) == "Font('Droid Sans Fallback Regular')"
+    assert repr(f) == "Font('Droid Sans Fallback Regular')"
 
 def test_add_ink_annot():
     import math
@@ -253,7 +253,7 @@ def test_get_text_dict():
         assert not wt
 
 def test_font():
-    font = pymupdf.Font()
+    font = pymupdf.Font(ordering=0)
     print(repr(font))
     bbox = font.glyph_bbox( 65)
     print( f'bbox={bbox!r}')
-- 
2.51.0.544.g305ed5581e

